### PR TITLE
Start linting

### PR DESCRIPTION
**Pull request by christianbundy**
Continued the cleanup from before, lots of manual tinkering but everything still looks right. I haven't tested this (other than `npm test`), but this removes a *lot* of dead code. Here are the things I was scared of and didn't touch:

- [`eqeqeq`](https://eslint.org/docs/rules/eqeqeq): Ideally we'd be using `===` instead of `==`, but I'm not familiar enough to know where that's trivial and where it's going to be a giant hairball
- [`import/no-extraneous-dependencies`](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-extraneous-dependencies.md): ESLint *really* wants us to install everything we `require()`, so requiring Electron in `main.js` complains that Electron shouldn't be a devDependency. Maybe it's correct? I'm not familiar enough.
- [`no-param-reassign`](https://eslint.org/docs/rules/no-param-reassign): didn't have time to change these, although I might have a clever fix for it.
- [`no-use-before-define`](https://eslint.org/docs/rules/no-use-before-define): basically just organizing functions so that they only call functions later in the file -- I'm pretty sure a computer could do this pretty well but my human brain is bad at it.

Let me know if you see anything funky!

### CODE REVIEW

File: inject.js [hunk 1]
```diff
@@ -77,34 +89,21 @@ exports.create = function (api) {
...
   }
 };
 
 function locationId(location) {
-  if (location.page === 'chess') return JSON.stringify({ page: 'chess' });
-  if (isChessMsg(location)) return JSON.stringify({ page: 'chess' });
+  if (location.page === 'chess' || isChessMsg(location)) {
+    return JSON.stringify({ page: 'chess' });
+  }
+  throw new Error('Unexpected location ID: ', location);
```
Comment (Happy0) [inline inject.js]: I think for `depject` to work, this has to return undefined if it doesn't handle the location (to signal the next handler should be tried.)

This stops patchbay starting as the error is thrown immediately.

File: ui/game/PieceGraveyard.js [hunk 2]
```diff
@@ -60,12 +65,12 @@ module.exports = (
   function renderPiecesForColour(colour) {
     let pieces = [];
 
-    for (const pieceName in materialDiff[colour]) {
+    Object.keys(materialDiff[colour]).forEach((pieceName) => {
```
Comment (Happy0) [inline ui/game/PieceGraveyard.js]: This throws an error upon opening a game now.

```
TypeError: Cannot convert undefined or null to object
```

Maybe we can do something more null safe here?

**Comment by Happy0** [comment]
I also seen a:

```
ReferenceError: otherPlayer is not defined
    at Object.keys.forEach (/home/happy0/projects/ssb-chess/ssb_ctrl/game.js:203:29)
```

**Review by Happy0 — CHANGES_REQUESTED** [review]

**Comment by Happy0** [comment]
Thanks for getting this underway, btw :)

**Comment by christianbundy** [comment]
Thanks for taking the time to give me feedback! I pushed some new code that seems reasonable in every way, *except* I had to switch the `++` and `--` logic for the piece graveyard. At first glance I'm not sure I completely understand the logic behind the code, but the original code was resulting in negative integers being passed to `R.repeat()`. Any idea why that might be? The fact that flipping the logic fixed the error is weird.

---

Oops, nevermind. Apparently I'd originally replaced `--` with `+=` and `++` with `-=`, which *of course* caused errors. I can play chess with `electron main.js` now!

---

Scratch that. I can open the app with `electron main.js` but it's currently not letting me play any moves. Hmm.

---

Another edit! It looks like I'm unable to make moves when using `electron main.js`, and the green circles don't show up showing my possible moves, but when I use the code in Patchbay everything seems to work perfectly. Any idea what might be happening there?

**Comment by Happy0** [comment]
I've checked out your branch and I'll see if I can reproduce the electron moves issue :) (once I have someone to move against :P)

**Comment by Happy0** [comment]
@christianbundy I can move fine when I'm on your branch (and using `./node_modules/electron/dist/electron main.js`).

You getting any errors in the developer console?

**Comment by christianbundy** [comment]
No errors, weird! I'll update if I notice anything else funky, thanks for the merge. :deciduous_tree:

**Pull request merged**